filter/patterns_storage_test.go

```diff
@@ -49,65 +55,73 @@ func TestProcessIncomingMetric(t *testing.T) {
 		Convey("For plain metric", func() {
-			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used 12 1234567890"))
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used 12 1234567890"), time.Hour)
 			So(matchedMetrics, ShouldBeNil)
 			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
 		})
 
 		Convey("For tag metric", func() {
-			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used;tag1=val1 12 1234567890"))
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used;tag1=val1 12 1234567890"), time.Hour)
 			So(matchedMetrics, ShouldBeNil)
 			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
 		})
 
 		Convey("For plain metric which has the same pattern like name tag in tagged pattern", func() {
-			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("tag.metric 12 1234567890"))
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("tag.metric 12 1234567890"), time.Hour)
 			So(matchedMetrics, ShouldBeNil)
 			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
 		})
 
 		Convey("For tagged metric which body matches plain metric trigger pattern", func() {
-			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("plain.metric;tag1=val1 12 1234567890"))
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("plain.metric;tag1=val1 12 1234567890"), time.Hour)
 			So(matchedMetrics, ShouldBeNil)
 			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
 		})
 
 		Convey("For plain metric which matches to tagged pattern which contains only name tag", func() {
-			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("name.metric 12 1234567890"))
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("name.metric 12 1234567890"), time.Hour)
 			So(matchedMetrics, ShouldBeNil)
 			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
 			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
 		})
+
+		Convey("For too old metric should miss it", func() {
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used 12 123"), time.Hour)
```

Comment:
> 2022-06-20 12:19:46.685 Scheduler INF metric disk.used is too old module=Scheduler


It's not OK to see it in test log. Let's increase logger level to warn for example.

Reply:
yep, done